NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -256,3 +415,517 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
+
+    Parameters
+    ----------
+    corsika_evt : NuRadio event object
+            use read_CORSIKA7() to create the event object containing the CoREAS output
+    """
+
+    def __init__(self, corsika_evt):
+        self.electric_field_on_sky = None
+        self.efield_times = None
+        self.obs_positions_geo = None
+        self.obs_positions_vxB_vxvxB = None
+        self.empty_efield = None
+        self.max_coreas_efield = None
+        self.star_radius = None
+        self.geo_star_radius = None
+
+        self.corsika_evt = corsika_evt
+        self.sim_station = corsika_evt.get_station(0).get_sim_station()
+        self.shower = corsika_evt.get_first_sim_shower()  # there should only be one simulated shower
+        self.star_shape_initilized = False
+        self.efield_interpolator_initilized = False
+        self.fluence_interpolator_initilized = False
+        self.zenith = self.shower.get_parameter(shp.zenith)
+        self.azimuth = self.shower.get_parameter(shp.azimuth)
+
+        self.cs = coordinatesystems.cstrafo(self.zenith, self.azimuth, self.shower[shp.magnetic_field_vector])
+
+        self.initialize_star_shape()
+
+    def initialize_star_shape(self):
+        """
+        Initializes the star shape pattern for interpolation, e.g. creates the arrays with the observer positions
+        in the shower plane and the electric field.
+        """
+        obs_positions_geo = []
+        obs_positions_onsky = []
+        electric_field_on_sky = []
+        efield_times = []
+
+        for j_obs, efield in enumerate(self.sim_station.get_electric_fields()):
+            obs_positions_geo.append(efield.get_position())
+            obs_positions_onsky.append(efield.get_position_onsky())
+            electric_field_on_sky.append(efield.get_trace().T)
+            efield_times.append(efield.get_times())
+
+        # shape: (n_observers, n_samples, (eR, eTheta, ePhi))
+        self.electric_field_on_sky = np.array(electric_field_on_sky)
+        self.efield_times = np.array(efield_times)
+        self.sampling_rate = 1. / (self.efield_times[0][1] - self.efield_times[0][0])
+        self.obs_positions_geo = np.array(obs_positions_geo)
+        self.obs_positions_vxB_vxvxB = np.array(obs_positions_onsky)
+
+        self.max_coreas_efield = np.max(np.abs(self.electric_field_on_sky))
+        self.empty_efield = np.zeros_like(self.electric_field_on_sky[0,:,:])
+
+        self.star_radius = np.max(np.linalg.norm(self.obs_positions_vxB_vxvxB[:, :-1], axis=-1))
+        self.geo_star_radius = np.max(np.linalg.norm(self.obs_positions_geo[:-1, :], axis=0))
+        logger.info(f'Initialize star shape pattern for interpolation. The shower arrives at zenith={self.zenith/units.deg:.0f}deg, azimuth={self.azimuth/units.deg:.0f}deg with radius {self.star_radius:.0f}m in the shower plane and {self.geo_star_radius:.0f}m on ground. ')
+        self.star_shape_initilized = True
+
+    def get_sampling_rate(self):
```

Comment:
This function is redundant, as the `sampling_rate` attribute is already public. If the goal is to obscure it from the user, we should make it a property instead (ie make this a real getter function)

Reply:
I wanted to use it in readCoreasDetector(). e.g.  l214. electric_field.set_trace(smooth_res_efield.T, self.coreas_interpolator.get_sampling_rate())

What would be a neat way?

Reply:
Instead of calling `get_sampling_rate()`, you can simply write `self.coreas_interpolator.sampling_rate` . This is what I meant with the attribute being public. Having a getter that simply returns a public attribute (in contrast to for example checking if some other variable has already been set, like done in `get_empty_efield()` ) is confusing to me, and I think also not preferred in Python.